airflow/ti_deps/deps/base_ti_dep.py

```diff
@@ -81,6 +81,10 @@ def get_dep_statuses(self, ti, session, dep_context):
         :param dep_context: the context for which this dependency should be evaluated for
         :type dep_context: DepContext
         """
+        from airflow.ti_deps.dep_context import DepContext
```

Comment:
We should avoid imports at local scope. I think the previous implementation was OK but it seems you have some concerns?